examples/regression/2-advanced-analysis/plot_coverage_validity.py

```diff
@@ -0,0 +1,528 @@
+#
+# To begin, we propose to use ``delta=0.8`` and ``n_delta=6`` and compare
+# the coverage validity claim of the MAPIE class and the referenced class.
+
+# Parameters of the modelisation
+delta = 0.8
+n_calib = 6
+
+n_train = 1000
+n_test = 1000
+num_splits = 1000
+
+# Load toy Data
+n_all = n_train + n_calib + n_test
+data, target = make_regression(n_all, random_state=1)
+
+# Split dataset into training, calibration and validation sets
+X_train, X_cal_test, y_train, y_cal_test = train_test_split(
+    data, target, train_size=n_train, random_state=1
+)
+
+# Create a regression model and fit it to the training data
+model = DecisionTreeRegressor()
+model.fit(X_train, y_train)
+
+# Compute theorical bounds and exact coverage to attempt
+lower_bound = delta
+upper_bound = (delta + 1/(n_calib+1))
+upper_bound_2 = (delta + 1/(n_calib/2+1))
+exact_cov = (np.ceil((n_calib+1)*delta))/(n_calib+1)
+
+# Run the experiment
+empirical_coverages_ref = []
+empirical_coverages_mapie = []
+
+for i in range(1, num_splits):
+    # Compute empirical coverage for each trial with StandardConformalizer
+    conformalizer = StandardConformalizer(model, non_conformity_func, delta)
+    coverage = get_coverage_prefit(
+        conformalizer, X_cal_test, y_cal_test, delta, n_calib, random_state=i
+    )
+    empirical_coverages_ref.append(coverage)
+
+    # Compute empirical coverage for each trial with MapieRegressor
+    conformalizer = MapieRegressor(estimator=model, cv="prefit")
+    coverage = get_coverage_prefit(
+        conformalizer, X_cal_test, y_cal_test, delta, n_calib, random_state=i
+    )
+    empirical_coverages_mapie.append(coverage)
+
+cumulative_averages_ref = cumulative_average(empirical_coverages_ref)
+cumulative_averages_mapie = cumulative_average(empirical_coverages_mapie)
+
+# Plot the results
+fig, ax = plt.subplots()
+plt.plot(cumulative_averages_ref, alpha=0.5, label='SplitCP', color='r')
+plt.plot(cumulative_averages_mapie, alpha=0.5, label='MAPIE', color='g')
+
+plt.hlines(exact_cov, 0, num_splits, color='r', ls='--', label='Exact Cov.')
+plt.hlines(lower_bound, 0, num_splits, color='k', label='Lower Bound')
+plt.hlines(upper_bound, 0, num_splits, color='b', label='Upper Bound')
+
+plt.xlabel(r'Split Number')
+plt.ylabel(r'$\overline{\mathbb{C}}$')
+plt.title(r'$|D_{cal}| = $' + str(n_calib) + r' and $\delta = $' + str(delta))
+
+plt.legend(loc="upper right", ncol=2)
+plt.ylim(0.7, 1)
+plt.tight_layout()
+plt.show()
+
+
+##############################################################################
+# It can be seen that the two curves overlap, proving that both methods
+# produce the same results. Their effective coverage stabilizes between
+# the theoretical limits, always above the target coverage and converges
+# towards the exact coverage (i.e. expected according to the theory).
+
+
+##############################################################################
+# Experiment 2: Again but without fixing random_state
```

Comment:
I would only keep the experiment 2 where you fix the `random_state`.

Reply:
I have to keep this part, because the reader will no longer understand why in the next plot, the different methods no longer overlap when they did before.